client/extensions/wp-job-manager/components/setup/page-setup/index.jsx

```diff
@@ -3,38 +3,122 @@
  */
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import page from 'page';
 import { connect } from 'react-redux';
-import { formValueSelector, reduxForm } from 'redux-form';
+import { Field, formValueSelector, reduxForm } from 'redux-form';
 import { localize } from 'i18n-calypso';
 import { flowRight as compose } from 'lodash';
 
 /**
  * Internal dependencies
  */
+import { Steps } from '../constants';
 import Button from 'components/button';
 import CompactCard from 'components/card/compact';
 import ExternalLink from 'components/external-link';
 import FormFieldset from 'components/forms/form-fieldset';
+import FormInputValidation from 'components/forms/form-input-validation';
 import FormSettingExplanation from 'components/forms/form-setting-explanation';
-import ReduxFormTextInput from 'components/redux-forms/redux-form-text-input';
+import FormTextInput from 'components/forms/form-text-input';
 import ReduxFormToggle from 'components/redux-forms/redux-form-toggle';
 import SectionHeader from 'components/section-header';
+import { createPages } from '../../../state/setup/actions';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isCreatingPages, shouldGoToNextStep } from '../../../state/setup/selectors';
 
 const form = 'extensions.wpJobManager.pageSetup';
 
+const validate = ( values, { translate } ) => {
+	const errors = {};
+	const message = translate( 'Page name cannot be empty.' );
+
+	if ( values.createPostJob && ! values.postJobTitle.trim() ) {
+		errors.postJobTitle = message;
+	}
+
+	if ( values.createDashboard && ! values.dashboardTitle.trim() ) {
+		errors.dashboardTitle = message;
+	}
+
+	if ( values.createJobs && ! values.jobsTitle.trim() ) {
+		errors.jobsTitle = message;
+	}
+
+	return errors;
+};
+
+const PageRenderer = ( { disabled, explanation, input, meta } ) => {
+	const isError = !! ( meta.touched && meta.error );
+
+	return (
+		<div className="page-setup__page">
```

Comment:
If you went the route with adding `containerComponent` to `ReduxFormFieldset`, it would have worked pretty well here, too:
```jsx
const Container = props => <div className="page-setup__page" {...props />;
return (
  <ReduxFormFieldset
    containerComponent={ Container }
    name="jobsTitle"
    disabled={ ... }
    explanation={ ... }
  />
);
```
Just swapping `FormFieldset` for another element.

The `PageRenderer` is perfectly good, too and there is nothing negative about it. I'm just commenting to point out an alternative implemenation :)